pychromecast/discovery.py

```diff
@@ -325,7 +326,9 @@ def _poll_hosts(self):
                 # This host should not be polled
                 continue
 
-            device_status = get_device_info(host, timeout=30, context=self._context)
+            device_status = get_device_info(
+                host, timeout=self._timeout / len(known_hosts), context=self._context
+            )
```

Comment:
I don't understand the point of this change, `_poll_hosts` runs continuously from a worker thread, why does the timeout matter?

Reply:
`discover_chromecasts` waits for `DISCOVER_TIMEOUT` before returning chromecasts discovered. If that timeout is not divided between the hosts that need to be discovered, then no chromecasts are returned if the first takes a long time or times out.

Reply:
`discover_chromecasts` is really only useful for manual testing from console, and if that's what you're doing keeping track of chromecasts jumping to another IP-address seems moot.
Can you maybe explain your use case a bit further?

Reply:
Also, please move this change to a separate PR, let's keep this PR about fixing known_hosts for older chromecasts.

Reply:
Removed the timeout change.
I don't really have a use case for `discover_chromecasts`, I only used it for manual testing of the change because it's documented in readme.